Multisig/Presentation/UI Library/QRScanner/QRCodeScannerViewController.swift

```diff
@@ -0,0 +1,158 @@
+//
+//  ScannerViewController.swift
+//  Multisig
+//
+//  Created by Moaaz on 4/15/20.
+//  Copyright © 2020 Gnosis Ltd. All rights reserved.
+//
+
+import UIKit
+import AVFoundation
+
+protocol QRCodeScannerViewControllerDelegate {
+    func scannerViewControllerDidScan(_ code: String)
+}
+
+class QRCodeScannerViewController: UIViewController {
+    @IBOutlet weak var headerLabel: UILabel!
+    var previewLayer: AVCaptureVideoPreviewLayer!
+    var captureSession: AVCaptureSession!
+    var delegate: QRCodeScannerViewControllerDelegate?
+    var header: String?
+    
+    enum Strings {
+        static let cameraAlertTitle = NSLocalizedString("camera_title", comment: "")
+        static let cameraAlertMessage = NSLocalizedString("camera_message", comment: "")
+        static let cameraAlertCancel = NSLocalizedString("cancel", comment: "")
+        static let cameraAlertAllow = NSLocalizedString("settings", comment: "")
+        static let scannerNotSupportedTitle = NSLocalizedString("scanner_not_supported_title", comment: "")
+        static let scannerNotSupportedMessage = NSLocalizedString("scanner_not_supported_message", comment: "")
+        static let ok = NSLocalizedString("ok", comment: "")
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        headerLabel.text = header
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+
+        if (captureSession?.isRunning == false) {
+            captureSession.startRunning()
+        }
+    }
+
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+
+        if (captureSession?.isRunning == true) {
+            captureSession.stopRunning()
+        }
+    }
+    
+    @IBAction func closeButtonTouched(_ sender: Any) {
+        dismiss(animated: true, completion: nil)
+    }
+    
+    func setup() {
+        AVCaptureDevice.requestAccess(for: .video) { [weak self] (result)  in
+            if result {
+                DispatchQueue.main.async {
+                    self?.createScanner()
+                }
+            }
+            else {
```

Comment:
we did not discuss coding style before, but I would suggest for the consistency to put 'else' near the closing bracket of 'if' as you did it below in this file